### PR TITLE
Make Windows wheel

### PR DESCRIPTION
**Pull request by Huy-Ngo**
Since it is messy to install old distribution of MSVC I'll skip that part.

### CODE REVIEW

**Comment by McSinyx** [comment]
Welcome to Windows, where local setup is even harder than CI :laughing:  Tag me when it's ready for review, Imma unsubscribe to avoid checking it 20 times a day.

**Comment by Huy-Ngo** [comment]
Hmm this one even failed to generate code fully 1bb8159

**Comment by McSinyx** [comment]
1bb8159 actually worked, the error is with linkage (i.e. it compiled successfully):

    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\cibw\python\python.3.6.8\tools\libs /LIBPATH:c:\cibw\python\python.3.6.8\tools\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\um\x64" /EXPORT:PyInit_palace build\temp.win-amd64-3.6\Release\src\palace.obj C:/projects/openal-soft-1.20.1-bin/libs/Win64/OpenAL32.lib libalure2-NOTFOUND /OUT:build\lib.win-amd64-3.6\palace.cp36-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.6\Release\src\palace.cp36-win_amd64.lib
    LINK : fatal error LNK1181: cannot open input file 'libalure2-NOTFOUND.obj'

**Comment by Huy-Ngo** [comment]
Hmm should I revert the last commit then? It doesn't seem to be affecting anyway.

**Comment by Huy-Ngo** [comment]
What should we do about this linking problem then?

**Comment by McSinyx** [comment]
> What should we do about this linking problem then?

Short answer: we should trynna solve it.  Long answer: first check what's with libalure2-NOTFOUND.obj and see if anything affect the linkage.  My guess is that there's something wrong with the generated Alure2::alure2 target, since this is what installed:

    -- Install configuration: "Release"
    -- Installing: C:/Program Files (x86)/alure/lib/alure2.lib
    -- Installing: C:/Program Files (x86)/alure/bin/alure2.dll
    -- Installing: C:/Program Files (x86)/alure/lib/alure2_s.lib
    -- Installing: C:/Program Files (x86)/alure/lib/cmake/Alure2/Alure2Config.cmake
    -- Installing: C:/Program Files (x86)/alure/lib/cmake/Alure2/Alure2Config-release.cmake
    -- Installing: C:/Program Files (x86)/alure/include/AL/alure2.h
    -- Installing: C:/Program Files (x86)/alure/include/AL/alure2-aliases.h
    ...

The output of `cmake palace/` could help inspect what's going on.

**Comment by Huy-Ngo** [comment]
Erm... I think I should delete the on-failed scripts too.

File: .appveyor.yml [hunk 1]
```diff
@@ -22,3 +22,7 @@ install:
   - py -3 -m pip install cibuildwheel
 
 build_script: echo py -3 -m cibuildwheel --output-dir wheelhouse
+
+on_failure:
+  - echo "Failed build, here is the source code"
+  - type c:\projects\palace\src\palace.cpp | findstr "ChannelConfig SampleType Spatialize"
```
Comment (McSinyx) [inline .appveyor.yml]: I don't think this is any longer needed. otherwise LGTM.  Please pay attention to the commit message too.

Comment (Huy-Ngo) [reply .appveyor.yml]: Yea I thought so to.

**Pull request merged**